Task76 as a customer i would like to select products for purchase them

Ecomont opened the pull request.

Ecomont committed:
Add ProductsController

Ecomont committed:
Add data insertion for Brands and Products tables

Ecomont committed:
Add GetProductsForPurchasing and update comments/docs

Removed an unused SQL query from `dbo.Brands.data.sql`
that retrieved schema and table names for specific tables.

Ecomont committed:
Refactor Products API to use DTOs and add filtering

Updated `GetProductsForPurchasing` in `ProductsController.cs`:
- Added optional `colour` parameter for filtering.
- Changed return type to `ProductForPurchaseDTO` for better data encapsulation.
- Updated LINQ query to filter by `colour` and sort by `Colour`.

Introduced `ProductForPurchaseDTO` class:
- Added properties for `ProductId`, `Name`, `Colour`, and `Brand`.
- Included validation annotations for `Name` and `Colour`.
- Provided a constructor for initialization.

Commented out deprecated `GetProductsForPurchasingAll` method.

Refactored code to improve separation of concerns and adhere to API design best practices.

Ecomont committed:
Update product filtering and seed data for `Colour` and `Name`

Enhanced the `GetProductsForPurchasing` method in the `ProductsController` to support filtering by both `colour` and `name`. Updated the LINQ query to handle the new `name` parameter and ensure proper filtering logic.

Modified the `dbo.Products.data.sql` file to include meaningful `Colour` values for the `Jacket` and `Shirt` products, aligning with the updated filtering logic. Previously, the `Colour` column was set to `NULL`.

Ecomont committed:
Enhance product purchasing functionality

Updated `GetProductsForPurchasing` to include `Stock` and `City` in the response. Refactored LINQ query and adjusted DTO projection.

Added new constructor and properties (`Stock`, `City`) to `ProductForPurchaseDTO` with validation annotations.

Modified `dbo.Products.data.sql` to include sample data for users, payment methods, orders, and relationships between products and orders. Improved data structure to support new functionality.

alejandrogomezl commented on dbo.Brands.data.sql:
Delet it from root, you have this data in the other SQL

Ecomont replied:
I have already delete it

alejandrogomezl reviewed (changes requested):
Check filters, http error and delete files not used.
<img width="558" height="138" alt="image" src="https://github.com/user-attachments/assets/8128f31d-85a9-4ee1-8f7b-c726e79779ea" />

Ecomont committed:
Remove specific brand insertions from SQL script

Ecomont committed:
Remove null check for product.Name in filtering logic

The filtering logic in the `GetProductsForPurchasing` method of the `ProductsController` class has been updated. The null check for the `product.Name` property was removed before checking if it contains the `name` parameter. This change assumes that `product.Name` is never null, simplifying the condition.

Remove redundant null check for product.Name

Updated the filtering logic in the GetProductsForPurchasing
method of ProductsController by removing the unnecessary
null check for product.Name before checking if it contains
the name parameter. This assumes product.Name is not null.

Ecomont committed:
Filter out-of-stock products in GetProductsForPurchasing

Added a condition to the LINQ query in the GetProductsForPurchasing method within the ProductsController class to ensure that only products with a stock greater than zero are included in the results. This change prevents out-of-stock products from being displayed to users looking to purchase items.

Ecomont committed:
Merge branch 'development' into task76-as-a-customer-i-would-like-to-select-products-for-purchase-them

alejandrogomezl commented on src/AppForSEII2526.API/Controllers/ProductsController.cs:
Check teacher's solution

Ecomont committed:
Merge with the branch

alejandrogomezl reviewed (approved):
Well done
